### PR TITLE
GEF deal creation with bank object instead of bankId

### PR DESCRIPTION
**Pull request by ttbarnes**
Instead of GEF deals having: `deal.bankId` string, it now stores the whole bank object in the deal: `deal.bank`.

This is now the same as BSS - although BSS will be updated in another PR so the bank object is on the top level and has the same object name.

This change also simplifies a Central API 'get deal' endpoint, where it used to make an API call to get the bank from `bankId`.

## Changes to Portal "all deals" filters

Currently we have an "all deals" query made from Portal  for 2 views - 1 for BSS, 1 for GEF. Filters are passed to the query which check that the bank id matches the users's bank id. This was being made inside the portal-api by default regardless of the `dealType`. This has been moved to the UI side, so it's just an additional filter passed from the UI and now handles the current differences between BSS and GEF data structure.

In the next PR this can be reverted because the data structure will be the same.

Also, there are 2 failing Portal APi tests (99.9% sure these are flaky) that have been disabled (Agreed with the team). They assert BSS deals sorting order by updated timestamp. This works in the real world, but fails a lot in the API tests. This will all change in 2 PRs time so feels like wasted time trying to fix (already spent time on it with no success).

### CODE REVIEW

File: gef-ui/server/controllers/application-details/index.test.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {
 } from '.';
 import api from '../../services/api';
 
-import mocks from '../mocks';
+import MOCKS from '../mocks';
```
Comment (Zainzzkk) [inline gef-ui/server/controllers/application-details/index.test.js:7]: this lowercase may have been me sorry 😂

**Review by Zainzzkk — COMMENTED** [review]
LGTM but 2 failing tests

File: gef-ui/server/controllers/application-details/index.test.js (hunk 1 above)
Comment (ttbarnes) [reply gef-ui/server/controllers/application-details/index.test.js:7]: np :D

File: gef-ui/server/controllers/mocks/index.js [hunk 2]
```diff
@@ -85,45 +86,50 @@ const MockApplicationResponseDraft = () => ({
...
-      lastName: 'Bloggs',
-      id: 1235,
+const MockApplicationResponseSubmitted = () => {
+  const now = new Date();
+  const yesterday = sub(now, { days: 1 });
+
+  return {
+    _id: '1234',
+    exporter: {},
+    bank: { id: 'BANKID' },
```
Comment (abhi-markan) [inline gef-ui/server/controllers/mocks/index.js:98]: Should we not also have `bankInternalRefName` inside `bank` object?

File: gef-ui/server/utils/helpers.js [hunk 3]
```diff
@@ -413,11 +413,15 @@ const areUnissuedFacilitiesPresent = (application) => {
    from date of submission
 */
 const facilityIssueDeadline = (submissionDate) => {
-  // converts to timestamp from epoch - '+' to convert from str to int
-  const date = new Date(+submissionDate);
-  const deadlineDate = add(new Date(date), { months: 3 });
+  if (submissionDate) {
+    // converts to timestamp from epoch - '+' to convert from str to int
+    const date = new Date(+submissionDate);
```
Comment (abhi-markan) [inline gef-ui/server/utils/helpers.js:418]: Instead of using unary plus, shall we use native `Number()` or `parseInt()`?

File: trade-finance-manager-api/src/v1/controllers/acbs.controller.js [hunk 4]
```diff
@@ -36,8 +36,8 @@ const clearACBSLog = async () => {
 
 const createACBS = async (deal) => {
   // Add bank's full details so we can reference partyUrn in function
-  const bankId = deal.dealSnapshot.bankId
-    ? deal.dealSnapshot.bankId
+  const bankId = deal.dealSnapshot.bank
+    ? deal.dealSnapshot.bank.id
     : deal.dealSnapshot.details.owningBank.id;
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/controllers/acbs.controller.js:41]: `owingBank.id` will it be a separate PR where it will appear under `bank` object?

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: gef-ui/server/controllers/mocks/index.js (hunk 2 above)
Comment (ttbarnes) [reply gef-ui/server/controllers/mocks/index.js:98]: `bankInternalRefName` is like a "name of the deal" for the bank. Could move inside bank object, but this is big change

File: gef-ui/server/utils/helpers.js (hunk 3 above)
Comment (ttbarnes) [reply gef-ui/server/utils/helpers.js:418]: @Zainzzkk ? I did not add this, just moved the comment

File: trade-finance-manager-api/src/v1/controllers/acbs.controller.js (hunk 4 above)
Comment (ttbarnes) [reply trade-finance-manager-api/src/v1/controllers/acbs.controller.js:41]: Yes will be seperate PR

File: gef-ui/server/controllers/mocks/index.js (hunk 2 above)
Comment (abhi-markan) [reply gef-ui/server/controllers/mocks/index.js:98]: Okay cool can leave it for another PR

File: gef-ui/server/utils/helpers.js (hunk 3 above)
Comment (abhi-markan) [reply gef-ui/server/utils/helpers.js:418]: @Zainzzkk happy for me to do this in my current PR?

**Review by abhi-markan — APPROVED** [review]

File: gef-ui/server/utils/helpers.js (hunk 3 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/helpers.js:418]: Yeah that is fine, easier for others to understand whats happening if parseInt actually
